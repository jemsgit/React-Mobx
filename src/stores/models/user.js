import { types } from 'mobx-state-tree';

const data = {
    isAuth: false,
    login: 'User'
}

export const User = types.model({
    isAuth: types.boolean,
    login: types.string
}).actions(self => {
    function setAuth(auth) {
        self.isAuth = auth
    }
    return {
        setAuth
    }
}).views(self => ({
    get loginator(){
        return self.login;
    }
}));