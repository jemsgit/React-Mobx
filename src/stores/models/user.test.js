import { getSnapshot, onSnapshot, onPatch } from 'mobx-state-tree';
import { reaction } from 'mobx';
import { User } from './user';

it('should be unauthorized', () => {
    let item = User.create({
        isAuth: false,
        login: '213'
    });
    let patches = [];

    onPatch(item, (patch) => {
        patches.push(patch)
    })

    item.setAuth(true);
    expect(patches).toMatchSnapshot();
})

it('should return login', ( ) => {
    let item = User.create({
        isAuth: false,
        login: '213'
    });

    expect(item.loginator).toBe('213');
    let chan = 0;
    reaction(() => item.isAuth, () => chan++);
    expect(chan).toBe(0);
    item.setAuth(true);
    expect(chan).toBe(1);
}) 