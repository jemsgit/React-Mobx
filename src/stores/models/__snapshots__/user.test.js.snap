// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be unauthorized 1`] = `
Array [
  Object {
    "op": "replace",
    "path": "/isAuth",
    "value": true,
  },
]
`;
